robot_framework/process.py

```diff
@@ -1,8 +1,242 @@
 """This module contains the main process of the robot."""
 
+import os
+import json
+from io import BytesIO
+from datetime import datetime
+from email.message import EmailMessage
+import smtplib
+
 from OpenOrchestrator.orchestrator_connection.connection import OrchestratorConnection
+from itk_dev_shared_components.graph import authentication, mail
+import openpyxl
+from openpyxl.worksheet.worksheet import Worksheet
+import pyodbc
+
+from robot_framework import config
 
 
 def process(orchestrator_connection: OrchestratorConnection) -> None:
     """Do the primary process of the robot."""
     orchestrator_connection.log_trace("Running process.")
+
+    excel_file = get_email_attachment(orchestrator_connection)
+    wb = read_excel_file(excel_file)
+
+    get_address_changes(wb)
+    calculate_difference(wb)
+
+    send_email(wb, orchestrator_connection)
+
+    clear_email_folder(orchestrator_connection)
+
+
+def get_email_attachment(orchestrator_connection: OrchestratorConnection) -> BytesIO:
+    """Find the email in the "AKBO" email folder and get the attached Excel file.
+
+    Args:
+        orchestrator_connection: The connection to OpenOrchestrator.
+
+    Raises:
+        ValueError: If something doesn't seem right with the email.
+
+    Returns:
+        The Excel file as a BytesIO object.
+    """
+    graph_creds = orchestrator_connection.get_credential(config.GRAPH_API)
+    graph_access = authentication.authorize_by_username_password(graph_creds.username, **json.loads(graph_creds.password))
+
+    emails = mail.get_emails_from_folder("itk-rpa@mkb.aarhus.dk", "Indbakke/AKBO", graph_access)
+    if len(emails) != 1:
+        raise ValueError(f"Unexpected number of emails found: {len(emails)}")
+
+    attachments = mail.list_email_attachments(emails[0], graph_access)
+    if len(attachments) != 1:
+        raise ValueError(f"Unexpected number of attachments found: {len(attachments)}")
+
+    if not attachments[0].name.endswith(".XLSX"):
+        raise ValueError(f"Unexpected attachment found: {attachments[0].name}")
+
+    return mail.get_attachment_data(attachments[0], graph_access)
+
+
+def read_excel_file(excel_file: BytesIO) -> openpyxl.Workbook:
+    """Read the excel sheet and filter away rows that don't meet the criteria.
+
+    Args:
+        excel_file: The file to read.
+
+    Returns:
+        The Excel workbook after filtering.
+    """
+    wb = openpyxl.load_workbook(excel_file)
+    ws: Worksheet = wb.active
+
+    # Check column names and get indices
+    columns = [c.value for c in ws[1]]
+    columns.index('CPR')
+    columns.index('Beløb')
```

Comment:
Do you mean to assign these indexes to variables? Otherwise, delete these lines.

Reply:
They are there to check if the columns exists. It's just to fail earlier rather than later